yarn-project/circuits.js/src/cbind/circuits.gen.ts

```diff
@@ -0,0 +1,954 @@
+/* eslint-disable */
+// GENERATED FILE DO NOT EDIT, RUN yarn remake-bindings
+import { Buffer } from 'buffer';
+import { callCbind } from './cbind.js';
+import { CircuitsWasm } from '../wasm/index.js';
+type MsgpackAddress = Buffer;
+type MsgpackFr = Buffer;
+type MsgpackFq = Buffer;
+type MsgpackProof = Buffer;
+import {
+  Address,
+  Fr,
+  Fq,
+  G1AffineElement,
+  NativeAggregationState,
+  NewContractData,
+  FunctionData,
+  OptionallyRevealedData,
+  PublicDataTransition,
+  PublicDataRead,
+  CombinedAccumulatedData,
+  PrivateHistoricTreeRoots,
+  CombinedHistoricTreeRoots,
+  ContractDeploymentData,
+  TxContext,
+  CombinedConstantData,
+  KernelCircuitPublicInputs,
+  Proof,
+  VerificationKeyData,
+  PreviousKernelData,
+} from './types.js';
+import { TupleOf, mapTuple, mapValues } from '@aztec/foundation/serialize';
+export interface MsgpackG1AffineElement {
+  x: Buffer;
+  y: Buffer;
+}
+export interface IG1AffineElement {
+  x: Fq;
+  y: Fq;
+}
+
+export function toG1AffineElement(o: MsgpackG1AffineElement): G1AffineElement {
+  if (o.x === undefined) {
+    throw new Error('Expected x in G1AffineElement deserialization');
+  }
+  if (o.y === undefined) {
+    throw new Error('Expected y in G1AffineElement deserialization');
+  }
+  return new G1AffineElement(Fq.fromBuffer(o.x), Fq.fromBuffer(o.y));
+}
+
+export function fromG1AffineElement(o: G1AffineElement): MsgpackG1AffineElement {
+  if (o.x === undefined) {
+    throw new Error('Expected x in G1AffineElement serialization');
+  }
+  if (o.y === undefined) {
+    throw new Error('Expected y in G1AffineElement serialization');
+  }
+  return {
+    x: o.x.toBuffer(),
+    y: o.y.toBuffer(),
+  };
+}
+
+export interface MsgpackNativeAggregationState {
+  P0: MsgpackG1AffineElement;
+  P1: MsgpackG1AffineElement;
+  public_inputs: Buffer[];
+  proof_witness_indices: number[];
+  has_data: boolean;
+}
+export interface INativeAggregationState {
```

Comment:
Good call. Originally I had more plans for them, barely realized I didn't use them anymore. Technically the fromNativeAggregationState could have taken an `o: INativeAggregationState`, but there isn't much use there as the interfaces just then refer to classes, so it's not some independent form. I will stop emitting these

Reply:
The original plan FTR was to have classes both implement INativeAggregationState and have from(INativeAggregationState) method. But instead I decided to explicitly generate constructor calls and member conversions, so instead its 'do you break circuits.gen.ts' instead of 'do you still implement this interface'. So yeah they can be removed now as they are redundant with the implicit class checking the from methods do